### ISSUE TITLE
OptionalChecker false-positive

### ISSUE THREAD

**Issue by ambition-consulting** [typetools/checker-framework#1959]
Is this a configuration error on my side? 

```
private A isOk(){
	final Optional<A> a = Optional.ofNullable((A) null);
	if (a.isPresent())
		return a.get();
	else
		throw new IllegalStateException();
}

// Wrong error: [method.invocation.invalid] call to <X>orElseThrow(java.util.function.Supplier<? extends X>) not allowed on the given receiver.
private A shouldBeOk() {
	return Optional.ofNullable((A) null).orElseThrow(IllegalStateException::new);
}

private class A{}
```

**Comment by mernst** [comment]
Can you please produce a complete test case (a complete, compilable Java file)?

Does this happen when you run the Checker Framework from the command line?
If so, this is not about the Optional Checker, but about the Maven plugin.

**Comment by mernst** [comment]
I'm closing this because we haven't received a reproducible test case.
If anyone can provide one, we will re-open the issue.

**Comment by ambition-consulting** [comment]
Sorry, not much time left these days. Will try to reproduce the next time I can work on the OptionalChecker.
